Modules/Filtering/ImageFilterBase/include/itkCastImageFilter.hxx

```diff
@@ -147,12 +147,12 @@ CastImageFilter<TInputImage, TOutputImage>::DynamicThreadedGenerateDataDispatche
 
   inputIt.GoToBegin();
   outputIt.GoToBegin();
+  OutputPixelType value{ Self::PixelFactory<OutputPixelType>::New(componentsPerPixel) };
```

Comment:
The pixel factory is not needed. Just do the following:
```
OutputPixelType value = outputIt.Get();
```
I believe the above should be a "deep copy" but please test the output in the test.

Reply:
Will do.